ugc/types.hpp

```diff
@@ -140,6 +140,50 @@ inline std::string DebugPrint(Text const & text)
   return os.str();
 }
 
+struct KeyboardText
+{
+  KeyboardText() = default;
+  KeyboardText(std::string const & text, uint8_t const deviceLang,
+               std::vector<uint8_t> const & keyboardLangs)
+    : m_text(text), m_deviceLang(deviceLang), m_keyboardLangs(keyboardLangs)
+  {
+  }
+  KeyboardText(Text const & text) : m_text(text.m_text), m_deviceLang(text.m_lang) {}
+  KeyboardText & operator=(KeyboardText const & text) = default;
+  KeyboardText & operator=(Text const & text)
+  {
+    m_text = text.m_text;
+    m_deviceLang = text.m_lang;
+    return *this;
+  }
+
+  DECLARE_VISITOR(visitor(m_text, "text"), visitor.VisitLang(m_deviceLang, "lang"),
+                  visitor.VisitLangs(m_keyboardLangs, "kbd_langs"))
+
```

Comment:
"keyboardLangs"?

Reply:
it's our naming agreement with server side